docs/source/create/materials/procedural-shaders.rst

```diff
@@ -0,0 +1,443 @@
+    	materialData: JSON.stringify({
+    		materials: {
+    			"model": "hifi_shader_simple"
+    			"procedural": {
+    			  	"version": 3,
+    			  	"shaderUrl": "https://docs.projectathena.dev/_static/resources/Proceduralv3.fs"
+    			}
+    		}
+    	})
+    });
+
+You must specify the material "model" as ``hifi_shader_simple`` and provide a shader link. To provide a fragment shader, set ``fragmentShaderURL`` (or ``shaderUrl``). To provide a vertex shader, set ``vertexShaderURL``.
+
+^^^^^^^^^^^^^^^
+Shader Template
+^^^^^^^^^^^^^^^
+
+When you learn about shaders for other applications, the shader may have a function like ``main()`` that is run first. By contrast, Project Athena has a specific function name that must be called. Which function is used depends on which version of the shader you use.
+
+As shaders were developed, features for them evolved a bit over time. As a result, there are several shader versions, and each version has a different call signature. **Versions 1 and 2** are the oldest, and will still work. **Versions 3 and 4** are the newest and expose more features. Version 4 provides for per-fragment positions, however it is also the most expensive. Therefore it is recommended to use Version 3 if that extra feature from Version 4 is not needed.
+
+A basic template for a shader will look something like this example::
+
+    // version 3
+    float getProceduralFragment(inout ProceduralFragment data) {
+        data.diffuse = vec3(0);
+        data.occlusion = 0;
+        data.roughness = 1;
+        data.emissive = _positionMS.xyz;
+        return 0; // "emissiveAmount", either <=0 or >0, suggest return 0 and use data.emissive
+    }
+
+The function ``getProceduralFragmentWithPosition()`` is the default main entry point for the fragment shader. Because shaders are always read by their compiler from top to bottom, this function must always be the last one in your shader code.
+
+This function also has the parameter ``ProceduralFragmentWithPosition``. This parameter is a data structure that contains entries that correspond to the usual Project Athena materials. The structure is described below::
+
+    struct ProceduralFragmentWithPosition {
+        vec3 position; // world space position
+        vec3 normal; // world space normal
+        vec3 diffuse;
+        vec3 specular;
+        vec3 emissive;
+        float alpha;
+        float roughness;
+        float metallic;
+        float occlusion;
+        float scattering;
+    };
+
+The default values for some of these are::
+
+    const float DEFAULT_ROUGHNESS = 0.9;
+    const float DEFAULT_SHININESS = 10.0;
+    const float DEFAULT_METALLIC = 0.0;
+    const vec3 DEFAULT_SPECULAR = vec3(0.1);
+    const vec3 DEFAULT_EMISSIVE = vec3(0.0);
+    const float DEFAULT_OCCLUSION = 1.0;
+    const float DEFAULT_SCATTERING = 0.0;
+    const vec3 DEFAULT_FRESNEL = DEFAULT_EMISSIVE;
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Shader Examples by Version
+^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
I don’t think this section is necessary.  each version is already explained below.  perhaps just an examples section at the bottom of the whole page?